Source/WebKit/WebProcess/GPU/graphics/gbm/RemoteGraphicsContextGLProxyGBM.cpp

```diff
@@ -45,43 +45,43 @@ class RemoteGraphicsLayerContentsDisplayDelegateGBM final : public GraphicsLayer
 
     virtual ~RemoteGraphicsLayerContentsDisplayDelegateGBM()
     {
-        m_proxy->setSwapBuffersFunction(nullptr);
     }
 
     void setDisplayBuffer(Ref<DMABufBuffer>&& displayBuffer, UnixFileDescriptor&& fenceFD)
     {
         std::swap(m_drawingBuffer, m_displayBuffer);
         m_displayBuffer = WTFMove(displayBuffer);
-        m_fenceFD = WTFMove(fenceFD);
+        pushDisplayBuffer(WTFMove(fenceFD));
     }
 
     void setDisplayBuffer(uint64_t bufferID, UnixFileDescriptor&& fenceFD)
     {
         if (m_drawingBuffer && m_drawingBuffer->id() == bufferID)
             std::swap(m_drawingBuffer, m_displayBuffer);
-        m_fenceFD = WTFMove(fenceFD);
+        pushDisplayBuffer(WTFMove(fenceFD));
     }
 
 private:
     explicit RemoteGraphicsLayerContentsDisplayDelegateGBM(bool isOpaque)
         : GraphicsLayerContentsDisplayDelegateTextureMapper(TextureMapperPlatformLayerProxy::create(TextureMapperPlatformLayerProxy::ContentType::WebGL))
         , m_isOpaque(isOpaque)
     {
-        m_proxy->setSwapBuffersFunction([this](TextureMapperPlatformLayerProxy& proxy) mutable {
-            if (!m_displayBuffer)
-                return;
-
-            OptionSet<TextureMapperFlags> flags = TextureMapperFlags::ShouldFlipTexture;
-            if (!m_isOpaque)
-                flags.add(TextureMapperFlags::ShouldBlend);
-            proxy.pushNextBuffer(CoordinatedPlatformLayerBufferDMABuf::create(Ref { *m_displayBuffer }, flags, WTFMove(m_fenceFD)));
-        });
+    }
+
+    void pushDisplayBuffer(UnixFileDescriptor&& fenceFD)
+    {
+        if (!m_displayBuffer)
+            return;
+
+        OptionSet<TextureMapperFlags> flags = TextureMapperFlags::ShouldFlipTexture;
+        if (!m_isOpaque)
+            flags.add(TextureMapperFlags::ShouldBlend);
+        m_proxy->pushNextBuffer(CoordinatedPlatformLayerBufferDMABuf::create(Ref { *m_displayBuffer }, flags, WTFMove(fenceFD)));
```

Comment:
...and finally this one here all do "the same". 

They push the next buffer, using a concrete buffer creation. They all do a similar alpha check to know when to add the `ShouldBlend` check, they need a fence...

Is there any room for (and / or benefit from) abstracting that? Maybe using some template method and having template specialisations in each class, so if we ever need to make changes it's easier to find all the specialisations?

Reply:
No, this is actually a preparation patch to stop using TextureMapperPlatformLayerProxy for delegated content and use a common delegate class.